request.js

```diff
@@ -304,9 +321,25 @@ TChannelRequest.prototype.resend = function resend() {
         return;
     }
 
-    peer.waitForIdentified(onIdentified);
+    var conn = peer.getInConnection(true);
+    var reqTimeout;
+    if (conn && conn.remoteName && !conn.closing) {
+        self.onIdentified(peer);
+    } else {
+        var waitForIdentifiedSlot = peer.waitForIdentified(onIdentified);
+
+        var now = self.channel.timers.now();
+        self.elapsed = now - self.start;
+        var timeout = self.timeout - self.elapsed;
+        var reqOp = new RequestOperation(
+            self, timeout, peer, waitForIdentifiedSlot
+        );
+        reqTimeout = self.channel.timeHeap.update(reqOp, now);
+    }
 
     function onIdentified(err) {
+        reqTimeout.cancel();
```

Comment:
some comment on why cancel needs to be called will be useful. isn't `reqTimeout` undefined at this point?